crates/rattler_repodata_gateway/src/gateway/query.rs

```diff
@@ -130,33 +140,57 @@ impl GatewayQuery {
             }
         }
 
-        // A list of futures to fetch the records for the pending package names. The main task
-        // awaits these futures.
+        let len = subdirs.len();
+        let mut result = vec![RepoData::default(); len];
+
+        // A list of futures to fetch the records for the pending package names. The
+        // main task awaits these futures.
         let mut pending_records = FuturesUnordered::new();
 
-        // The resulting list of repodata records.
-        let mut result = vec![RepoData::default(); subdirs.len()];
+        // Push the direct url queries to the pending_records.
+        for spec in direct_url_specs.clone() {
+            let gateway = self.gateway.clone();
+            pending_records.push(
+                async move {
+                    let url = spec
+                        .clone()
+                        .url
+                        .expect("direct url spec should always have an url");
+                    let query =
+                        DirectUrlQuery::new(url.clone(), gateway.package_cache.clone(), gateway.client.clone());
+                    query
+                        .execute()
+                        .await
+                        .map_err(|e| GatewayError::DirectUrlQueryError(url.to_string(), e))
+                        .map(|records| (0, vec![spec], records))
+                }
+                .boxed(),
```

Comment:
Should we maybe also verify that the name of the package matches with what we expected from the spec?

Reply:
For when someone fakes the name of the file?

Reply:
Added the check in https://github.com/mamba-org/rattler/pull/725/commits/d520aa0e6fca780e20151f854c2dccc74e488dd